### ISSUE TITLE
Why there is a `nextTick` in createRuleTester?

### ISSUE THREAD

**Issue by fa93hws** [stylelint/stylelint#4120]
Hi, I am currently looking for having a custom jest rule tester.
However, the `nextTick` at https://github.com/stylelint/stylelint/blob/master/lib/testUtils/createRuleTester.js#L120 block my implementation.
Does anyone know what's the reason why it is necessary here?

In jest
```
function foo() {
  describe('a test', () => {
    it('it', () => {
      expect(1).toBe(1);
    });
  });
}

process.nextTick(foo);
```
Doesn't work😰
Jest will be terminated as it can't find anything to test..

**Comment by evilebottnawi** [comment]
How it is break, please provide example

**Comment by fa93hws** [comment]
> How it is break, please provide example

Currently I implemented it as 
``` typescript
function assertEquality(processCss: Promise<RuleTesterResult[]>, context: RuleTesterContext) {
  const describeFn = context.only ? describe.only : describe;
  describeFn(context.caseDescription, () => {
    it(context.completeAssertionDescription, done => {
      processCss.then((comparisons) => {
        comparisons.forEach(({ actual, expected }) => {
          expect(actual).toEqual(expected);
        });
        done();
      }).catch(done);
    })
  })
}

const testRule = createRuleTester(assertEquality);
```

Any test actually fails because of the `nextTick` in https://github.com/stylelint/stylelint/blob/master/lib/testUtils/createRuleTester.js#L120
It would be the same as having 
``` javascript
function foo() {
  describe('a test', () => {
    it('it', () => {
      expect(1).toBe(1);
    });
  });
}

process.nextTick(foo);
```
only in a test file to jest

**Comment by hudochenkov** [comment]
I'm not sure why we have `createRuleTester` API and if anyone uses it, to be honest.

You can take a look at https://github.com/stylelint/jest-preset-stylelint. It's the same [Jest configuration](https://github.com/stylelint/stylelint/blob/master/jest-setup.js) we use in our repository, but extracted into separate package.

**Comment by fa93hws** [comment]
Thanks! Will have a look.
I remember `createRuleTester` is used in the tape binding of the rule tester, which is the only tester listed in the official document..

**Comment by fa93hws** [comment]
Hey wait... `createRuleTester` API is the one that was written on the document...
https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rule-testers.md

**Comment by fa93hws** [comment]
Is there any update?

**Comment by hudochenkov** [comment]
Have you seen [`createRuleTester.test.js`](https://github.com/stylelint/stylelint/blob/master/lib/testUtils/__tests__/createRuleTester.test.js)? There `createRuleTester` is used specifically with Jest.

**Comment by fa93hws** [comment]
> Have you seen [`createRuleTester.test.js`](https://github.com/stylelint/stylelint/blob/master/lib/testUtils/__tests__/createRuleTester.test.js)? There `createRuleTester` is used specifically with Jest.

Hi! I think we are talking about the same thing.
This issue is about `process.nextTick` https://github.com/stylelint/stylelint/blob/master/lib/testUtils/createRuleTester.js#L120 which prevent it to be used in jest as I mentioned here https://github.com/stylelint/stylelint/issues/4120#issuecomment-503158028. The jest won't run anything because there is nothing to test at the first tick.

**Comment by hudochenkov** [comment]
There is another implementation for Jest https://github.com/keplersj/jest-runner-stylelint.

I'm inclining to close this issue.

**Comment by vankop** [comment]
@fa93hws last commit is dated to 2016 =) my suggestion is to use runner provided by @hudochenkov  ☝️

@hudochenkov our tests can not find this problem due to run in `it` case already, see https://github.com/stylelint/stylelint/blob/master/lib/testUtils/__tests__/createRuleTester.test.js

**Comment by fa93hws** [comment]
Oh cool, thanks
